### ISSUE TITLE
EDAMUserException: ENML_VALIDATION parameter=Attribute \"id\" must be declared for element type \"map\"."

### ISSUE THREAD

**Issue by HaleTom** [robert7/nixnote2#98]
#### Expected
Sync completes without error

#### Actual
```
INFO  2019-04-12 13:19:53.579 src/html/enmlformatter.cpp:344 enml-cleanup: ===== rebuilding note ENML
INFO  2019-04-12 13:19:53.585 src/html/enmlformatter.cpp:308 enml-cleanup: tidy DONE: diagnostics: "line 1 column 1 - Warning: missing <!DOCTYPE> declaration; line 1 column 44 - Warning: inserting missing 'title' element; Info: Document content looks like XHTML5; Tidy found 2 warnings and 0 errors!; ; "
INFO  2019-04-12 13:19:53.610 src/html/enmlformatter.cpp:308 enml-cleanup: tidy DONE: diagnostics: "line 1 column 1 - Warning: missing <!DOCTYPE> declaration; line 1 column 44 - Warning: inserting missing 'title' element; Info: Document content looks like XHTML5; Tidy found 2 warnings and 0 errors!; ; "
INFO  2019-04-12 13:19:53.610 src/html/enmlformatter.cpp:443 enml-cleanup: ===== finished rebuilding note ENML
ERROR 2019-04-12 13:20:56.826 src/communication/communicationmanager.cpp:499 Exception stacktrace:
ERROR 2019-04-12 13:20:56.833 src/communication/communicationerror.cpp:86 "EDAMUserException[ENML_VALIDATION]: EDAMUserException: ENML_VALIDATION parameter=Attribute \"id\" must be declared for element type \"map\"."
ERROR 2019-04-12 13:22:07.734 src/communication/communicationmanager.cpp:499 Exception stacktrace:
ERROR 2019-04-12 13:22:07.736 src/communication/communicationerror.cpp:86 "EDAMUserException[ENML_VALIDATION]: EDAMUserException: ENML_VALIDATION parameter=Attribute \"id\" must be declared for element type \"map\"."
```

Possibly related:  #95 

## Steps to reproduce the problem (or [log file](https://github.com/robert7/nixnote2/wiki/HowTo---Add-log-file-to-the-github-issue))

  1. Press sync button

Could the title of the note be included in the error message?  I'm not sure which one is causing the issue.

## Specifications

Version: 2.1.1-f5e206dd

**Comment by robert7** [comment]
did you tried the newest version 2.1.3?

**Comment by robert7** [comment]
so pls try the newest version
and I need the log file with DEBUG level *incl.* the "files" subdirectory (so I can see the text of the notes)
https://github.com/robert7/nixnote2/wiki/HowTo---Add-log-file-to-the-github-issue
just make sure you don't include any private data in the example notes

**Comment by HaleTom** [comment]
Logs from `Version: 2.1.3-bbbf0a1a`:

```
DEBUG 2019-04-18 17:10:25.245 src/communication/communicationmanager.cpp:503 reportError
ERROR 2019-04-18 17:10:25.246 src/communication/communicationmanager.cpp:513 Exception stacktrace:
ERROR 2019-04-18 17:10:25.251 src/communication/communicationerror.cpp:86 "EDAMUserException[ENML_VALIDATION]: EDAMUserException: ENML_VALIDATION parameter=Attribute \"id\" must be declared for element type \"map\"."
DEBUG 2019-04-18 17:10:25.251 src/utilities/debugtool.cpp:71 *** Dumping Note ***
DEBUG 2019-04-18 17:10:25.251 src/utilities/debugtool.cpp:216 "guid":"61d4a2c9-ce27-4ea8-a3a3-848a24159240":
DEBUG 2019-04-18 17:10:25.251 src/utilities/debugtool.cpp:224 "active":true:
DEBUG 2019-04-18 17:10:25.251 src/utilities/debugtool.cpp:216 "title":"A Gift of Fire: Social, Legal, and Ethical Issues for Computing Technology (4th Edition)":
DEBUG 2019-04-18 17:10:25.251 src/utilities/debugtool.cpp:214 Attachment: #10 "note-content.xml"
DEBUG 2019-04-18 17:10:25.254 src/utilities/debugtool.cpp:261 "contentHash":"b232ce5b09082a458074206042c34f26":
DEBUG 2019-04-18 17:10:25.254 src/utilities/debugtool.cpp:238 "contentLength":446412:
DEBUG 2019-04-18 17:10:25.254 src/utilities/debugtool.cpp:245 "created":1479707392000:
DEBUG 2019-04-18 17:10:25.254 src/utilities/debugtool.cpp:245 "updated":1554892360526:
DEBUG 2019-04-18 17:10:25.255 src/utilities/debugtool.cpp:238 "USN":852:
DEBUG 2019-04-18 17:10:25.255 src/utilities/debugtool.cpp:216 "source":"web.clip":
DEBUG 2019-04-18 17:10:25.255 src/utilities/debugtool.cpp:216 "sourceURL":"https://www.amazon.com/Gift-Fire-Ethical-Computing-Technology/dp/0132492679?tag=bisafetynet-20":
DEBUG 2019-04-18 17:10:25.255 src/utilities/debugtool.cpp:216 "sourceApplication":"mobile.android":
DEBUG 2019-04-18 17:10:25.255 src/utilities/debugtool.cpp:117 resource #1/23
DEBUG 2019-04-18 17:10:25.255 src/utilities/debugtool.cpp:128 guid="78c827b7-b7b2-4b43-a3c9-73732963d7a7", noteGuid="61d4a2c9-ce27-4ea8-a3a3-848a24159240", mime="image/jpeg", width=128, height=160, duration=885, active=true, updateSequenceNum=(unset)
```

The note mentioned maps to `files/0010-note-content.xml`.

Nothing personal in the included `files/*`.
[nixnote-issue-98.tar.gz](https://github.com/robert7/nixnote2/files/3093902/nixnote-issue-98.tar.gz)

Very nice wiki instructions on how to report issues BTW.

**Comment by HaleTom** [comment]
Given the above, can you please reopen?

**Comment by robert7** [comment]
@HaleTom sorry I'm bit busy now, but I take a look later

**Comment by robert7** [comment]
@HaleTom problem involving "map" html tag is fixed in 2.1.4-a, please test and let me know, if you find any other situation involving html cleanup/sync problem

**Comment by HaleTom** [comment]
Cheers, thanks a lot!
